CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+
+We welcome feedback from the user community.
+This document is intended to help guide users on best practices for becoming involved with OpenMDAO.
+
+## Contents
+
+- [Questions about OpenMDAO](##Questions-about-OpenMDAO)
+- [Reporting bugs](##Reporting-bugs)
+- [Proposing new features](##Proposing-new-features)
+- [Submitting pull requests](##Submitting-pull-requests)
+- [License](##License)
+
+## Questions about OpenMDAO
+
+We encourage questions regarding OpenMDAO to be asked on [stackoverflow.com](https://stackoverflow.com)
+with the [OpenMDAO](https://stackoverflow.com/questions/tagged/openmdao) tag.
+The developer team monitors the OpenMDAO tag at stackoverflow fairly closely,
+so this is the best way to get timely feedback.
+
+## Reporting Bugs
+
+To report a bug, use the [New Issue](https://github.com/OpenMDAO/OpenMDAO/issues/new/choose) form
+on GitHub, and select `Get Started` on the "Bug Report" option. The bug report template will guide
+you through the needed information, including a description of the erroneous behavior and some
+example code demonstrating how to replicate the error.
+
+## Proposing New Features
+
+OpenMDAO has a formal process for proposing, discussing and approving changes that effect the behavior
+of the code or add new features. Such a proposal is dubbed a "POEM" and the process is explained in detail
+[here](https://github.com/OpenMDAO/POEMs/blob/master/POEM_000.md). In short, you are asked to submit a
+pull request to the [POEMs](https://github.com/OpenMDAO/POEMs) repository with a document (and optional
+supporting files) describing the motivation for proposed change, the desired outcome and the suggested
+API changes.
+
+## Submitting Pull Requests
+
+Code contributions to OpenMDAO are welcome, however all pull requests are expected to address an open
+[issue](https://github.com/OpenMDAO/OpenMDAO/issues). If there is no existing issue related to the code
+change you wish to submit, then one should be created via the
+[New Issue](https://github.com/OpenMDAO/OpenMDAO/issues/new/choose) form. New issues are usually associated
+with a [bug report](##Reporting-bugs) or [POEM](##Proposing-new-features) as described above, however
+you also have the option to submit an issue regarding an unexpected or undocumented behavior of the code
+or a problem with the documentation, testing or other related aspects of OpenMDAO.
+
+In order for a pull request to be accepted, it must include one or more unit tests verifying that the new
+implementation works while not causing any unrelated tests to fail. It must also comply with OpenMDAO coding
+standards, which can be checked by running the tests in the `openmdao.code_review` module.
+Our coding standards generally align with the [PEP8](https://peps.python.org/pep-0008/) and
+[numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) standards.
+
+It is recommended that you install OpenMDAO with `pip install -e .[all]` in your development environment
```

Comment:
I wonder if it makes more sense to use the `master` branch from a cloned repository rather than pip installing openmdao, since the code contribution has to come through git anyway.

Reply:
Not sure I understand this comment... The intent of this statement is that they should test their local development branch before submitting a pull request.. so this is saying that they need to install openmdao from that branch and test it...